contrib/check-config.sh

```diff
@@ -158,7 +158,10 @@ echo -n '- '
 if [ "$(stat -f -c %t /sys/fs/cgroup 2> /dev/null)" = '63677270' ]; then
 	echo "$(wrap_good 'cgroup hierarchy' 'cgroupv2')"
 else
-	cgroupSubsystemDir="$(awk '/[, ](cpu|cpuacct|cpuset|devices|freezer|memory)[, ]/ && $3 == "cgroup" { print $2 }' /proc/mounts | head -n1)"
+	cgroupSubsystemDir=$(grep -e cpu -e cpuacct -e cpuset -e device -e freezer -e memory /proc/mounts | while read -r _ mpoint type _; do [ "$type" = cgroup ] && {
```

Comment:
This changes the actual implementation in a pretty subtle way, and IMO pretty drastically decreases the readability/maintainability of it.

I'm really curious what environment you're in that has a `/bin/sh` but doesn't have an `awk` implementation -- can you elaborate?  Even BusyBox has a pretty good `awk` implementation. :sweat_smile:

Reply:
I reckon the readability matter depends on how much a power user of awk one is. :-)

I tested back and forth and thought I had understood the current awk command as find those entries in /proc/mounts and match the ones with the type croup, then send them to head to just use the first of them. So I did, as you write, a subtle change in that I search the entries and stop searching having found the first match that has the cgroup type. But I reckon the outcome should stay correct? (Can there be such entries where the type is not cgroup, by the way?)

As you guessed, I didn't do this change just for the kicks, but on the edge device ([Axis Q1615-LE MkIII](https://www.axis.com/products/axis-q1615-le-mk-iii)I use there was no awk to be found. I could certainly build one and provide it just to run this script, but thought it easier to avoid the awk dependency.

Reply:
As for the readability part, perhaps replacing the efficient

```shell
[ "$type" = cgroup ] && {
...
}
```

expression with an

```shell
if [ "$type" = cgroup ]; then
...
fi
```

block instead would make it more approachable for maintainers?

Reply:
I did an update with that now, please see what you think about it.